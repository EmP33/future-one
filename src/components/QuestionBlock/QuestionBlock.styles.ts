import styled from "styled-components";

export const Wrapper = styled.div<{ showDetails: boolean }>`
  border-bottom: 1px solid var(--color-text);
  display: grid;

  .question-answer {
    display: grid;
    grid-template-columns: 1fr max-content;
    row-gap: 0.5rem;
    height: 100%;
    padding: 1.5rem 0;
    cursor: pointer;
    transition: all 0.1s linear;
    background: ${({ showDetails }) =>
      showDetails ? "var(--color-primary)" : "transparent"};
    color: var(--color-secondary);

    &:hover {
      background-color: var(--color-primary);
      color: var(--color-secondary);
    }

    & span {
      display: block;
    }

    & h3 {
      font-weight: bold;
      font-size: 28px;
      grid-row: 2;
    }
    & button {
      grid-row: span 2;
      display: flex;
      align-self: center;
      align-items: center;
      justify-content: center;
      height: 50px;
      width: 50px;
      font-size: 1.6rem;
      border-radius: 50%;
      border: 3px solid var(--color-green);
      background: transparent;
      color: var(--color-green);
      transform: ${({ showDetails }) =>
        showDetails ? "rotate(0)" : "rotate(-90deg)"};
      transition: all 0.15s linear;
      cursor: pointer;

      @media only screen and (max-width: 600px) {
        height: 45px;
        width: 45px;
      }
    }
  }

  .hidden-content {
    display: block;
    color: var(--font-color-1);
    max-height: ${({ showDetails }) => (showDetails ? "100px" : 0)};
    overflow: hidden;
    transition: max-height 0.3s ease-in, padding 0.4s linear;

    @media only screen and (max-width: 600px) {
      overflow: auto;
    }

    & > p {
      padding: 0.5rem 0 2rem;
    }
  }
`;
