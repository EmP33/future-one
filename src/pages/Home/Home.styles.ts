import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;

  .hero {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    height: 70vh;
    align-items: center;

    @media only screen and (max-width: 1200px) {
      height: 80vh;
      grid-template-columns: 1fr 1.5fr;
    }

    @media only screen and (max-width: 900px) {
      grid-template-columns: 1fr;
      margin-top: 3rem;
    }

    @media only screen and (max-width: 600px) {
      height: auto;
      margin-top: 6rem;
    }

    &-action {
      display: grid;
      align-items: center;
      align-content: center;
      justify-items: start;
      width: 450px;
      row-gap: 1.25rem;

      @media only screen and (max-width: 1200px) {
        width: 100%;
      }

      @media only screen and (max-width: 900px) {
        grid-row: 2;
      }

      & h1 {
        font-size: 3rem;
        font-weight: bold;

        @media only screen and (max-width: 1200px) {
          font-size: 2.5rem;
        }

        & span {
          color: var(--color-green);
        }
      }

      & p {
        width: 80%;
        line-height: 1.5rem;

        @media only screen and (max-width: 900px) {
          width: 100%;
        }

        @media only screen and (max-width: 600px) {
          font-size: 15px;
        }
      }
    }
  }
`;
