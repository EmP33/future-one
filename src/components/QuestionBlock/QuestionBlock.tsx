import React, { useState } from "react";
import { Wrapper } from "./QuestionBlock.styles";
import { AiOutlinePlus, AiOutlineMinus } from "react-icons/ai";
import { IQuestion } from "../../data";

interface IProps {
  question: IQuestion;
}

const QuestionBlock: React.FC<IProps> = ({
  question: { id, answer, question },
}) => {
  const [showDetails, setShowDetails] = useState(false);

  return (
    <Wrapper showDetails={showDetails}>
      <div className="question-answer">
        <span>{id}.</span>
        <h3>{question}</h3>
        <button onClick={() => setShowDetails((prev) => !prev)}>
          {showDetails ? <AiOutlineMinus /> : <AiOutlinePlus />}
        </button>
      </div>
      <div className="hidden-content">
        <p>{answer}</p>
      </div>
    </Wrapper>
  );
};

export default QuestionBlock;
