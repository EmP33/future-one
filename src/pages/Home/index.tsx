import React from "react";
import Layout from "../../components/Layout/Layout";
import { OutlineButton, PrimaryButton } from "../../Global.styles";
import { Wrapper, Newsletter } from "./Home.styles";
import { hero } from "../../assets";
import QuestionBlock from "../../components/QuestionBlock/QuestionBlock";
import Quote from "../../components/Quote/Quote";
import { categories, questions, quote } from "../../data";
import CategoryBlock from "../../components/CategoryBlock/CategoryBlock";

const Home = () => {
  return (
    <Layout>
      <>
        <Wrapper>
          <section className="hero">
            <div className="hero-action">
              <h1>
                <span>Duis aute irure</span> dolor in reprehenderit
              </h1>
              <p>
                Nam at lectus urna duis convallis. Id semper risus in hendrerit
                gravida rutrum quisque non tellus. Elit eget gravida cum sociis
                natoque. Ultrices neque ornare aenean euismod elementum nisi
                quis eleifend. Arcu felis bibendum ut tristique et egestas.
              </p>
              <PrimaryButton size="big" to="/">
                Discover now
              </PrimaryButton>
            </div>
            <div className="hero__image-wrapper">
              <img src={hero} alt="smiling woman" width="100%" height="100%" />
            </div>
          </section>
          <section className="faq">
            <h2 className="faq__title">Lacinia quis vel eros tempor orci.</h2>
            <div className="faq-content">
              {questions.map((question) => (
                <QuestionBlock key={question.id} question={question} />
              ))}
            </div>
          </section>
          <Quote quote={quote} />
          <section className="categories">
            <h2 className="categories__title">
              Massa tempor nec feugiat nisl pretium fusce
            </h2>
            <p>
              Pellentesque <strong>habitant</strong> morbi tristique senectus et
              netus et malesuada. <strong>Ipsum faucibus vitae</strong> aliquet
              nec ullamcorper sit amet risus nullam. Dictum varius duis at
              consectetur <strong>lorem donec</strong> massa sapien. Massa
              tempor nec feugiat nisl pretium fusce.
            </p>
            <p>Elit scelerisque mauris pellentesque?</p>
            <div className="categories-blocks">
              {categories.map((category) => (
                <CategoryBlock key={category.id} category={category} />
              ))}
            </div>
          </section>
        </Wrapper>
        <Newsletter>
          <div className="newsletter">
            <h2>Subscribe to our newsletter</h2>
            <div className="newsletter-action">
              <input type="text" placeholder="Company e-mail address" />
              <OutlineButton>Subscribe</OutlineButton>
            </div>
            <p>Chupa chups gummi bears shortbread candy</p>
          </div>
        </Newsletter>
      </>
    </Layout>
  );
};

export default Home;
