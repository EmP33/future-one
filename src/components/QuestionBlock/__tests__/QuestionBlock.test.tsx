import QuestionBlock from "../QuestionBlock";
import ShallowRenderer from "react-test-renderer/shallow";
import { IQuestion } from "../../../pages/Home";

const question: IQuestion = {
  question: "Example question",
  answer: "Example Answer",
  id: "01",
};

beforeEach(() => {
  // setup a DOM element as a render target
  let container = document.createElement("div");
  container.setAttribute("id", "modal-root");
  document.body.appendChild(container);
});

describe("<QuestionBlock />", () => {
  it("should render properly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<QuestionBlock question={question} key={question.id} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
