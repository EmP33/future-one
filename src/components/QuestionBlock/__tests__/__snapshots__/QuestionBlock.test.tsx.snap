// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionBlock /> should render properly 1`] = `
<styled.div
  showDetails={false}
>
  <div
    className="question-answer"
  >
    <span>
      01
      .
    </span>
    <h3>
      Example question
    </h3>
    <button
      aria-label="Expand"
      onClick={[Function]}
    >
      <AiOutlinePlus />
    </button>
  </div>
  <div
    className="hidden-content"
  >
    <p>
      Example Answer
    </p>
  </div>
</styled.div>
`;
